client/src/common/components/Members/components/MemberDetails.jsx

```diff
@@ -0,0 +1,316 @@
+import React, { Fragment, PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import classNames from 'classnames';
+
+import { RouterMatchPropType, UserPropType } from 'common/constants/propTypes';
+import { CloseIcon, InfoIcon, UserIcon } from 'assets/images/icons';
+import { getCurrentUser } from 'modules/authorization/model/selectors';
+import {
+  changeRole as changeRoleInCohort,
+  removeCohortMember
+} from 'modules/cohort/model/actions';
+import {
+  changeRole as changeRoleInList,
+  removeListMember
+} from 'modules/list/model/actions';
+import { Routes, UserRoles } from 'common/constants/enums';
+import { noOp } from 'common/utils/noOp';
+
+const infoText = {
+  [Routes.COHORT]: {
+    [UserRoles.OWNER]:
+      "Can edit, archive and delete this cohort. Can add, edit list's items and mark them as done. Can add, edit, archive and delete lists. Can add, remove members, and change their roles.",
+    [UserRoles.MEMBER]: "Can view lists, add and edit list's items."
+  },
+  [Routes.LIST]: {
+    [UserRoles.OWNER]:
+      "Can edit, archive and delete this list. Can add, edit list's items and mark them as done. Can add, remove members, and change their roles.",
+    [UserRoles.MEMBER]: "Can view, add and edit list's items."
+  }
+};
+
+class MemberDetails extends PureComponent {
+  constructor(props) {
+    super(props);
+    const { isOwner } = props;
+    this.state = {
+      isConfirmVisible: false,
+      isMemberInfoVisible: false,
+      isOwnerInfoVisible: false,
+      role: isOwner ? UserRoles.OWNER : UserRoles.MEMBER
+    };
+  }
+
+  handleConfirmationVisibility = () => {
+    const { isConfirmVisible } = this.state;
+    this.setState({ isConfirmVisible: !isConfirmVisible });
```

Comment:
change to: `this.setState(({ isConfirmVisible }) => ({ isConfirmVisible: !isConfirmVisible }));`